// Copyright (C) 2016 Nippon Telegraph and Telephone Corporation.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
// implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package l2;

message MACAddress {
    bytes address = 1;
}

message VLANID {
    uint32 id = 1;
}

message Address {
    uint32 flags = 1;
    uint32 station_flags = 2;
    MACAddress mac = 3;
    VLANID vid = 4;
    int64 port = 5;
    int64 mod_id = 6;
}
